### PR TITLE
Fix compiler errors in the tests related to c_char signedness differe…

### PR DESCRIPTION
**Pull request by sdroege**
…nces on different platforms

Fixes https://github.com/gtk-rs/gtk-rs-core/issues/166

----

CC @remilauzier: Please confirm that this actually fixes your test failures on ARM.

### CODE REVIEW

**Comment by sdroege** [comment]
Also needs to be backported to 0.14.

**Comment by sdroege** [comment]
And also need to figure out how to make clippy happy

```
error: casting integer literal to `u8` is unnecessary
   --> glib/src/char.rs:203:62
    |
203 |         assert_eq!(UChar(241 as c_uchar), unsafe { from_glib(241 as c_uchar) });
    |                                                              ^^^^^^^^^^^^^^ help: try: `241_u8`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast
```

**Review by bilelmoussaoui — APPROVED** [review]

**Comment by sdroege** [comment]
Should I maybe just silence this specific clippy warning for these tests, or does someone have a better idea?

**Comment by GuillaumeGomez** [comment]
CI is failing because of unnecessary casts.

File: glib/src/char.rs [hunk 1]
```diff
@@ -169,10 +169,10 @@ mod tests {
...
     #[test]
     fn converts_single_byte_chars() {
-        assert_eq!(Char::try_from(0 as char), Ok(Char(0_i8)));
-        assert_eq!(UChar::try_from(0 as char), Ok(UChar(0_u8)));
-        assert_eq!(UChar::try_from(255 as char), Ok(UChar(255_u8)));
-        assert_eq!(UChar::try_from('ñ'), Ok(UChar(241_u8)));
+        assert_eq!(Char::try_from(0 as char), Ok(Char(0 as c_char)));
+        assert_eq!(UChar::try_from(0 as char), Ok(UChar(0 as c_uchar)));
+        assert_eq!(UChar::try_from(255 as char), Ok(UChar(255 as c_uchar)));
+        assert_eq!(UChar::try_from('ñ'), Ok(UChar(241 as c_uchar)));
```
Comment (GuillaumeGomez) [inline glib/src/char.rs:175]: Maybe use `as _`, it might not complain about it.

Comment (sdroege) [reply glib/src/char.rs:175]: No:

```
warning: casting integer literal to `i8` is unnecessary
   --> glib/src/char.rs:172:55
    |
172 |         assert_eq!(Char::try_from(0 as char), Ok(Char(0 as _)));
    |                                                       ^^^^^^ help: try: `0_i8`
    |
```

Comment (sdroege) [reply glib/src/char.rs:181]: Silenced the warnings for now until someone has a better idea. The same would've also happened in the non-test code on ARM (or wherever a C `char` is unsigned).

Comment (GuillaumeGomez) [reply glib/src/char.rs:181]: I guess it's fine.

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**